src/user.ts

```diff
@@ -55,6 +57,8 @@ export class User {
     const changes: UserChanges = {
       ...syncProperty(this, 'name', userState.name),
       ...syncProperty(this, 'channelId', userState.channelId),
+      // -1 is returned to signal successful user registration deletion
+      ...syncPropertyWithMapper(this, 'userId', (v) => v === 4294967295 ? undefined : v , userState.userId),
```

Comment:
I don't understand why we need the `withMapper` variation here. Can't we just call
```ts
...syncProperty(this, 'userId', userState.userId === 4294967295 ? undefined : userState.userId),
```

Reply:
Also, please make the 4294967295 a constant, so that it doesn't become a magic number, e.g.
```ts
const MinusOneButUnsigned = 4294967295
```

Reply:
I found a better fix for this, using a getter/setter to handle the mapping